ec2-dash/dash_sqlite.py

```diff
@@ -0,0 +1,203 @@
+import csv
+import boto3
+import os
+
+# create a boto3 client for when the csv needs to be uplaoded to the s3 bucket
+s3 = boto3.client('s3')
+
+# import the confluent kafka consumer 
+c = create_kafka_consumer()
+
+conn = sqlite3.connect('./ec2-dash/dash_db.db') #create an sqlite database and establish a connection
+
+cursor = conn.cursor() #create a cursor to allow querying of the database
+
+def create_new_current_ride_table(cursor: sqlite3.Cursor, conn: sqlite3.Connection):
+    """Drops the previous table called current_ride and creates a new one. This function will be 
+    called immediately after a new user information dict is received
+
+    Args:
+        cursor (sqlite3.Cursor): The cursor that allows queries to be made to the database
+        conn (sqlite3.Connection): Enables the query to be committed
+
+    """
+
+    
+    cursor.execute("DROP TABLE IF EXISTS current_ride") #deletes the old current_ride table
+
+    # creates a new current_ride table 
+    cursor.execute("""
+    CREATE TABLE current_ride (
+        date TEXT, 
+        time TEXT,
+        duration INTEGER NOT NULL,
+        resistance INTEGER NOT NULL, 
+        heart_rate INTEGER NOT NULL,
+        rpm INTEGER NOT NULL, 
+        power TEXT NOT NULL
+    )
+    """)
+
+    conn.commit()
+
+def add_entry_to_table(cursor: sqlite3.Cursor,conn: sqlite3.Connection, entry: dict):
+    """Adds an entry to the current_ride table. Called every time a new log is received 
+    from the kafka consumer
+
+    Args:
+        cursor (sqlite3.Cursor): The cursor that allows queries to be made to the database
+        conn (sqlite3.Connection): Enables the query to be committed
+        entry (dict): The combined log entry for this event
+    """
+
+    # query sent to the table current_ride to insert into table 
+    cursor.execute(f"""
+    INSERT INTO current_ride (
+        date, time, duration, resistance, heart_rate, rpm, power
+    )
+    VALUES (
+        "{entry['date']}", "{entry['time']}", 
+        {entry['duration']}, {entry['resistance']}, 
+        {entry['heart_rate']}, {entry['rpm']}, 
+        "{entry['power']}"
+    )
+    """)
+
+    conn.commit()
+
+def most_recent_ride_to_csv(cursor: sqlite3.Cursor):
+    """Writes the most recent ride to a csv file ready to be pushed to an s3 bucket
+
+    Args:
+        cursor (sqlite3.Cursor): The cursor that allows queries to be made to the database
+    """
+    # retrieving all the data from the sqlite table
+    cursor.execute("""
+    SELECT * FROM current_ride
+    """)
+    most_recent_ride = cursor.fetchall()
+
+    # writes the data to a csv file 
+    with open("ec2-dash/most_recent_ride.csv", "w") as f:
+        csv_f = csv.writer(f)
+        csv_f.writerow(['date', 'time', 'duration', 'resistance', 'heart_rate', 'rpm', 'power'])
+        csv_f.writerows(most_recent_ride)
+
+def create_user_info_table(cursor: sqlite3.Cursor, conn: sqlite3.Connection, user_info: dict):
```

Comment:
Could make the name uniform with the other table recreation function, i.e. recreate_user_info_table and recreate_current_ride_table